src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate from = LocalDate.parse(dateFrom, formatter).minusDays(1);
+        LocalDate till = LocalDate.parse(dateTo, formatter).plusDays(1);
+        String[][] splittedData = new String[data.length][];
+        String[] result = new String[names.length];
+        for (int i = 0; i < data.length; i++) {
+            splittedData[i] = data[i].split(" ");
+        }
+
+        for (int j = 0; j < names.length; j++) {
+            StringBuilder stringBuilder = new StringBuilder();
+            int totalInRange = 0;
+            stringBuilder.append(names[j]).append(" - ");
+            for (int i = 0; i < splittedData.length; i++) {
```

Comment:
```suggestion
            for (int i = 0; i < data.length; i++) {
            	String[] spslittedData = data.split(" ");
```
тоді можна буде видалити цикл на 13-15 строках